src/windows/sid.rs

```diff
@@ -0,0 +1,153 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use std::{fmt::Display, str::FromStr};
+
+use winapi::{
+    shared::{
+        sddl::{ConvertSidToStringSidW, ConvertStringSidToSidW},
+        winerror::ERROR_INSUFFICIENT_BUFFER,
+    },
+    um::{
+        errhandlingapi::GetLastError,
+        securitybaseapi::{CopySid, EqualSid, GetLengthSid, IsValidSid},
+        winbase::{LocalFree, LookupAccountSidW},
+        winnt::{SidTypeUnknown, LPWSTR, PSID},
+    },
+};
+
+use crate::sys::utils::to_str;
+
+#[doc = include_str!("../../md_doc/sid.md")]
+#[derive(Clone, Debug)]
+pub struct Sid {

```

Comment:
This type shouldn't be public as it doesn't have an equivalent in other platforms.

Reply:
Unfortunately, if I mark it as `pub(crate)` it doesn't compile:

```plain
error[E0446]: crate-private type `Sid` in public interface
   --> src\common.rs:674:13
    |
674 |             type Target = $type;
    |             ^^^^^^^^^^^ can't leak crate-private type
...
738 |         uid!(crate::windows::Sid);
    |         ------------------------- in this macro invocation
    |
   ::: src\windows\sid.rs:21:1
    |
21  | pub(crate) struct Sid {
    | --------------------- `Sid` declared as crate-private
    |
    = note: this error originates in the macro `xid` which comes from the expansion of the macro `uid` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Since `Uid` implements `Deref` for the internal type, said internal type must be public.

Reply:
Ah right. Then please make `account_name` a `pub(crate)` method at least so it's not usable outside of the crate.